airflow/providers/amazon/aws/hooks/eks.py

```diff
@@ -596,3 +597,6 @@ def fetch_access_token_for_cluster(self, eks_cluster_name: str) -> str:
 
         # remove any base64 encoding padding:
         return "k8s-aws-v1." + base64_url.rstrip("=")
+
+    def get_waiter(self, waiter_name):
+        return EksBotoWaiter(client=self.conn).waiter(waiter_name)
```

Comment:
Just a thought: "Is it possible to make it part of base AWS hook?"

Reply:
Maybe, but then you'd have to pass it the waiter model every time too, right?

Reply:
It just an idea.

It could be some parameter of hook and if it None than use predefined model for hook (e.g. from waiters directory).

Reply:
On the other hand a lot of boto3 clients doesn't have waiters so it might be add additional complexity if it included in base hook

Reply:
Maybe possible to work some magic.... if every service's custom waiter model config is in ~/aws/waiters/{service}.json then the service name could be extracted from the hook

Reply:
so a  basehook solution might look like

```
def get_waiter(self):
    client_name = self,conn.something
    config_path = Path(__file__).with_name(f"{client_name}.json").resolve()
            with open(config_path) as config_file:
                self.default_config = json.load(config_file) 
    return BaseBotoWaiter(client=client_name, model_config=self.default_config)
```

Reply:
I think I got this figured out and it's even cleaner.  I am also adding some unit tests and need to update the README to match so I'll have an update maybe tonight or tomorrow.

get_waiter now lives in `airflow.providers.amazon.aws.hooks.base_aws.AwsGenericHook.get_waiter` and no change needs to be made tot he service hook to implement it.  :+1:   The code is a little more complicated than before, but new additions are easier so that's a net plus.

Reply:
Just one additional thing by default there are limited files which included in resulting wheel/sdist package, and json files do not add by default.

So we also need updates in this file:
https://github.com/apache/airflow/blob/5fad8a4a450de8388e4488523670d5f10f4f35b1/dev/provider_packages/MANIFEST_TEMPLATE.in.jinja2#L25-L30

For avoid the situation which happen with k8s provider: https://github.com/apache/airflow/pull/27451

Reply:
Ohhhhh, thank you.  I was unaware of that.    I had a chat with Vincent yesterday and he suggested another change so I should have this updated today.  Sorry for the delay.

Reply:
The file you linked at `dev/provider_packages/MANIFEST_TEMPLATE.in.jinja2` has a message in it saying

```
# IF YOU WANT TO MODIFY IT, YOU SHOULD MODIFY THE TEMPLATE
# `MANIFEST_TEMPLATE.py.jinja2` IN the `provider_packages` DIRECTORY
```

but there is no file by that name.  Do you know if that's an old warning that should be removed/ignored?

```
ferruzzi:~/workplace/airflow$ find . -name MANIFEST_TEMPLATE*
./dev/provider_packages/MANIFEST_TEMPLATE.in.jinja2
```

Reply:
This comment added to generated `MANIFEST.in` into the provider package.

I thought it some mistake and it should be `MANIFEST_TEMPLATE.in.jinja2` instead of `MANIFEST_TEMPLATE.py.jinja2`.

cc: @potiuk

Reply:
I changed `MANIFEST_TEMPLATE.in.jinja2`.  If that's not right, I can fix it.  I'll need to rebase before this gets merged anyway.